zebra-consensus/src/parameters.rs

```diff
@@ -0,0 +1,36 @@
+//! The consensus parameters for each Zcash network.
+//!
+//! Some consensus parameters change based on network upgrades. Each network
+//! upgrade happens at a particular block height. Some parameters have a value
+//! (or function) before the upgrade height, at the upgrade height, and after
+//! the upgrade height. (For example, the value of the reserved field in the
+//! block header during the Heartwood upgrade.)
+//!
+//! Typically, consensus parameters are accessed via a function that takes a
+//! `Network` and `BlockHeight`.
+
+use zebra_chain::block::BlockHeaderHash;
+use zebra_chain::{Network, Network::*};
+
+/// Returns the previous block hash for the genesis block in `network`.
+pub fn genesis_previous_block_hash(network: Network) -> BlockHeaderHash {
+    // All current networks use all-zeroes for the parent of the genesis block.
+    // (In Bitcoin, `null` is `[0; 32]`.)
+    //
+    // TODO: make this function const when feature(const_if_match) stabilises.
+    match network {
+        Mainnet | Testnet => BlockHeaderHash([0; 32]),
+    }
+}
```

Comment:
I think this is fine to hardcode, because there will be no new networks to support, let alone ones that don't make the same null choice as Bitcoin does.

Reply:
zcashd has a `regtest` network, I'm not sure whether we'll want to support it for network integration testing. But it also uses an all-zeroes genesis parent hash.

When you say hard-code, do you mean as a constant in this module? That's fine with me, I'll make that change.

Reply:
Turned into a constant in 9050d02e32218c70800ae8e538d12a4d354d89c1.

Reply:
... which makes the code a lot simpler, because the checkpoint verifier doesn't need to store the network any more. (It just passes it to the checkpoint list.)